nomos-services/time/src/backends/ntp/async_client.rs

```diff
@@ -21,21 +25,34 @@ pub enum Error {
 
 #[cfg_attr(feature = "serde", cfg_eval::cfg_eval, serde_with::serde_as)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-#[derive(Copy, Clone)]
+#[derive(Debug, Copy, Clone)]
 pub struct NTPClientSettings {
     /// NTP server requests timeout duration
     #[cfg_attr(feature = "serde", serde_as(as = "MinimalBoundedDuration<1, NANO>"))]
-    timeout: Duration,
-    /// NTP server socket address
-    address: SocketAddr,
+    pub timeout: Duration,
+    /// Local socket address to listen for responses
+    /// The bound IP behaves like a server listening IP. This means for WAN
+    /// requests you probably want to bind to `0.0.0.0`. For LAN requests
+    /// you can simply bind to the local IP.
+    pub local_socket: SocketAddr,
```

Comment:
User shouldn't care what port the local socket has, similar to how we don't care about it when making tcp requests. `local_socket` address could be removed, or changed to `interface` and expect `IpAddr` if we would like to allow users to request (for example) ntp over vpn or other public ip that they have.